### ISSUE TITLE
[Feature Request] Multi-account support

### ISSUE THREAD

**Issue by viren-nadkarni** [etspaceman/kinesis-mock#377]
Kinesis-Mock supports namespacing over regions which was introduced in https://github.com/etspaceman/kinesis-mock/pull/187

It would be very useful to build on that and allow namespacing over AWS accounts. This can be based on the Access Key ID retrieved from the Authorization header.

Some possible approaches:
- Extract the account ID from access key ID <https://awsteele.com/blog/2020/09/26/aws-access-key-format.html>
- Use the face value of access key ID for namespacing but use `AWS_ACCOUNT_ID` for all outputs (this approach is taken by DynamoDB Local)

**Comment by etspaceman** [comment]
I agree that this would be a valuable improvement. I may not get to this for a while. But if anyone wants to take a crack at it, I encourage it!

**Comment by etspaceman** [comment]
@viren-nadkarni This code snippet does not seem to work for all access key ID values per the blog post. I'm not sure we want to use this approach.
